docs/designs/conceptual-versioning.md

```diff
@@ -0,0 +1,474 @@
+
+4. All files will be built in one round, which will bring a better performance.
+
+##### For content reader
+
+1. Support viewing page with specific version by query `?view={moniker}`
+
+2. Support showing Toc with different version.
+
+3. [Empty page](#42-empty-page) should be handled.
+
+#### 1.2 Out of scope
+
+1. Not support specific version reference, including link and xref.
+
+    > If user reference another file by link/Uid with different monikerRange, the resolve result of this reference can not guaranteed.
+
+    Sample:
+
+    ```txt
+    |- articles/
+    |    |- folder1/
+    |    |   |- a.md
+    |    |- folder2/
+    |    |   |- b.md
+    |    |- folder3/
+    |        |- b.md
+    |- config.yml
+    ```
+
+    the content of `config.yml` is
+    ```yml
+    name: dotnet
+    content: "articles/**.{md,yml}"
+    monikerRange:
+        "articles/folder1/**.{md,yml}": "netcore-1.0"
+        "articles/folder2/**.{md,yml}": "netcore-1.0"
+        "articles/folder3/**.{md,yml}": "netcore-2.0"
+    routing:
+        "articles/folder1/": "articles/"
+        "articles/folder2/": "articles/"
+        "articles/folder3/": "articles/"
+    monikerDefinition: "https://api.docs.com/monikers/"
+    ```
+
+    If content writer want to reference `folder3/b.md` in `folder1/a.md` by link `'[B](../folder3/b.md)'`, when the reader click the link [B]() on page `{host}/{docset-base-path}/articles/a?view=netcore-1.0`, they will jump to page `{host}/{docset-base-path}/articles/b?view=netcore-1.0`, which is the output of file `folder2/b.md`.
+
+2. Not support doing bookmark validation inside different versioned zones.
+
+    Sample: if there is two conceptual file `articles/a.md`
+
+    ```markdown
+    ---
+    monikerRange: netcore-1.0 || netcore-2.0
+    ---
+
+    ::: moniker range="netcore-1.0"
+    ## heading1
+    ::: moniker-end
+
+    ::: moniker range="netcore-2.0"
+    ## heading2
+    ::: moniker-end
+    ```
+
+    If there is a link `'a#heading2'` in `articles/b.md` with moniker `netcore-1.0`, when user click the link, the bookmark `#heading2` will not work, but no warning will be reported in Phase 1.
+
+3. Not support generating static website. In phase 1, since docfx doesn't handle fallback case, there have to be host server.
+
+### 2 User Experience
+
+#### 2.1 Moniker Range
+
+In order to ease version configuration and content maintenance, [moniker range](./versioning.md#moniker-ranges) enable the writer to associate more than one moniker with content, but without needing to list monikers.
+
+We support three level moniker range setting:
+
+##### 2.1.1 Config file
+
+```yml
+name: dotnet
+content: "articles/**.{md,yml}"
+monikerRange:
+    "articles/v1.0/**.{md,yml}": "netcore-1.0"
+    "articles/v2.0/**.{md,yml}": "netcore-1.0 || >= netcore-1.0"
+routing:
+    "articles/v1.0/": "articles/"
+    "articles/v2.0/": "articles/"
```

Comment:
seems moniker and routing are paired to config, which brings complexity

Reply:
yes!
add to the open question.
keep current design before we have a better idea.

---
In reply to: [229264759](https://github.com/dotnet/docfx/pull/3522#discussion_r229264759) [](ancestors = 229264759)